url3f8d6440dd_huiwq1990_CognativeModel_tgz-pJ8-clarion_samples_HelloWorldJ8/src/sim/Hello.java

```diff
@@ -0,0 +1,429 @@
+				CombinedInputSpace.get(Personality.PrimaryDrives.AFFILIATION_AND_BELONGINGNESS)
+					.get(Drive.TypicalInputs.STIMULUS).setActivation(1);
+				//Specify the stimulus for the AUTONOMY drive.
+				CombinedInputSpace.get(Personality.PrimaryDrives.AUTONOMY)
+					.get(Drive.TypicalInputs.STIMULUS).setActivation(0);
+				System.out.print("You say, \"Good Morning\". ");
+			}
+			else if(random > 0.3)
+			{
+				//Specify the salutation to give as "Goodbye".
+				CombinedInputSpace.get("Salutations").get("GoodMorning").setActivation(0);
+				CombinedInputSpace.get("Salutations").get("GoodAfternoon").setActivation(1);
+				CombinedInputSpace.get("Salutations").get("GoodAfternoon").setActivation(0);
+				//Specify the stimulus for the AFFILIATON_AND BELONGINNESS drive.
+				CombinedInputSpace.get(Personality.PrimaryDrives.AFFILIATION_AND_BELONGINGNESS)
+					.get(Drive.TypicalInputs.STIMULUS).setActivation(0);
+				//Specify the stimulus for the AUTONOMY drive.
+				CombinedInputSpace.get(Personality.PrimaryDrives.AUTONOMY)
+					.get(Drive.TypicalInputs.STIMULUS).setActivation(1);
+				System.out.print("You say, \"Good Afternoon\". ");
+			}else{
+				//Specify the salutation to give as "Goodbye".
+				CombinedInputSpace.get("Salutations").get("GoodMorning").setActivation(0);
+				CombinedInputSpace.get("Salutations").get("GoodAfternoon").setActivation(0);
+				CombinedInputSpace.get("Salutations").get("GoodAfternoon").setActivation(1);
+				//Specify the stimulus for the AFFILIATON_AND BELONGINNESS drive.
+				CombinedInputSpace.get(Personality.PrimaryDrives.AFFILIATION_AND_BELONGINGNESS)
+					.get(Drive.TypicalInputs.STIMULUS).setActivation(0);
+				//Specify the stimulus for the AUTONOMY drive.
+				CombinedInputSpace.get(Personality.PrimaryDrives.AUTONOMY)
+					.get(Drive.TypicalInputs.STIMULUS).setActivation(1);
+				System.out.print("You say, \"Good Afternoon\". ");
+			}
+			if(i == 0)
+			{
+				/*Have the agent perceive the world. We are at trial 0, so no feedback has been 
+				 * provided yet.*/
+				Agent.perceive(CombinedInputSpace);
+			}
+			else
+			{
+				/*Have the agent perceive the world including the feedback for the last 
+				action it performed.*/
+				Agent.perceive(CombinedInputSpace, feedback);
+			}
+			//Have the agent choose an action.
+			ChosenAction = Agent.act();
+			if(ChosenAction.equalsID(Actions.get("Hello")))
+			{
+				//The agent said "Hello".
+				System.out.print("The agent said, \"Hello\". ");
+				if(CombinedInputSpace.get("Salutations").get("Hello").getActivation() == 1.0)
+				{
+					//The agent was right.
+					System.out.println("The agent was correct.");
+					//Record the agent's success.
+					CorrectCounter++;
+					//Give positive feedback.
+					feedback = 1;
+				}
+				else
+				{
+					//The agent was wrong.
+					System.out.println("The agent was incorrect.");
+					//Give negative feedback.
+					feedback = 0;
+				}
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 329:349, 350:370.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.